contracts/access/manager/IAccessManager.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IAuthority} from "./IAuthority.sol";
+import {IManaged} from "./IManaged.sol";
+import {Time} from "../../utils/types/Time.sol";
+
+interface IAccessManager is IAuthority {
+    enum AccessMode {
+        Custom,
+        Closed,
+        Open
+    }
+
+    // Structure that stores the details for a group/account pair. This structures fit into a single slot.
+    struct Access {
+        // Timepoint at which the user gets the permission. If this is either 0, or in the future, the group permission
+        // are not available. Should be checked using {Time-isSetAndPast}
+        uint48 since;
+        // delay for execution. Only applies to restricted() / relay() calls. This does not restrict access to
+        // functions that use the `onlyRole` modifier.
```

Comment:
Not reviewing the docs yet but I think this should be `onlyGroup` instead of `onlyRole`, right?